backend/apiPackage.ts

```diff
@@ -157,3 +157,74 @@ export async function extractMetadataFromZip(filebuffer: Buffer): Promise<apiSch
     ID: uuidv4(),
   }
 }
+
+// For: get package download
+export async function getPackageDownload(req: Request, res: Response) {
+	try {
+		const packageID = req.query.name;
+
+		if (packageID === undefined) {
+			return res.status(400).send('Package name or version is undefined');
+		}
+		const dbPackage = await prismaCalls.getPackage(packageID as string);
+		if (!dbPackage) {
+			return res.status(404).send('Package not found');
+		}
+		// transformed for the apiSchema structure
+		// can't use map unless I change apischema/prismaschema
+		const apiPackage: apiSchema.Package = {
+			metadata: {
+				Name: dbPackage.metadata.Name,
+				Version: dbPackage.metadata.Version,
+				ID: dbPackage.metadata.ID,
+			},
+			data: {
+				Content: dbPackage.data.Content,
+				URL: dbPackage.data.URL,
+				JSProgram: dbPackage.data.JSProgram,
+			},
+		};
+		return res.status(200).json(apiPackage);
+	} catch (error) {
+		return res.status(500).send(`Error in getPackageDownload: ${error}`);
+	}
+}
+
+// For: put package update
+export async function updatePackage(req: Request, res: Response) {
```

Comment:
idk if you added this or not but you have to check if the package exists or not. If it doesn't exist in the db, then return an error. (404  Package does not exist)